syntax = "proto3";
package product;

service ProductService {
  rpc GetProduct (ProductRequest) returns (ProductReply);
  rpc CreateProduct (CreateProductRequest) returns (ProductReply);
}

message ProductRequest {
  string id = 1;
}

message ProductReply {
  string id = 1;
  string name = 2;
  float price = 3;
}

message CreateProductRequest {
  string name = 1;
  float price = 2;
}